libs/remix-ui/debugger-ui/src/lib/debugger-ui.tsx

```diff
@@ -87,6 +87,18 @@ export const DebuggerUI = (props: DebuggerUIProps) => {
       })
     })
 
+    debuggerInstance.event.register('locatingBreakpoint', async (isActive) => {
+      setState(prevState => {
+        return { ...prevState, sourceLocationStatus: 'Locating breakpoint, this might take a while...' }
```

Comment:
I'm not sure if this setState will work as expected in a call back. You can define a new `useState` variable for `sourceLocationStatus` and update it directly.

Reply:
This works, I used the same method as line 106. let me know still if there's a better way to handle this warning message